src/components/NetworkSwitcher.tsx

```diff
@@ -527,18 +570,28 @@ function NetworksGrid({
 
   const dragging = useSharedValue<DraggingState | null>(null);
 
-  const pinnedHeight = useDerivedValue(() => Math.ceil(networks.value[Section.pinned].length / 2) * (ITEM_HEIGHT + GAP) - GAP);
-  const sectionsOffsets = useDerivedValue(() => ({
-    [Section.pinned]: { y: 0 },
-    [Section.unpinned]: { y: pinnedHeight.value + SEPARATOR_HEIGHT },
-  }));
-  const containerStyle = useAnimatedStyle(() => {
+  const sectionsOffsets = useDerivedValue(() => {
+    const pinnedHeight = Math.ceil(networks.value[Section.pinned].length / 2) * (ITEM_HEIGHT + GAP) - GAP;
+    return {
+      [Section.pinned]: { y: 0 },
+      [Section.separator]: { y: pinnedHeight },
+      [Section.unpinned]: { y: pinnedHeight + SEPARATOR_HEIGHT },
+    };
+  });
+  const containerHeight = useDerivedValue(() => {
+    const length = networks.value[Section.unpinned].length;
+    const paddingBottom = 32;
     const unpinnedHeight = isUnpinnedHidden.value
-      ? 0
-      : Math.ceil(networks.value[Section.unpinned].length / 2) * (ITEM_HEIGHT + GAP) - GAP + 32;
-    const height = pinnedHeight.value + SEPARATOR_HEIGHT + unpinnedHeight;
-    return { height: withTiming(height) };
+      ? length === 0
+        ? -SEPARATOR_HEIGHT + paddingBottom
+        : 0
+      : length === 0
+        ? ITEM_HEIGHT + paddingBottom
+        : Math.ceil((length + 1) / 2) * (ITEM_HEIGHT + GAP) - GAP + paddingBottom;
```

Comment:
got a couple of ideas to clean up a lot of this, and we are gonna need to tweak now that we are gonna support more networks, add scroll etc so gonna do it in a separate pr later